### PR TITLE
Clean up the View class

### PR DESCRIPTION
**Pull request by desertfire**
Summary:
- Remove used header files
- Remove unused functions
- Replace View::IrNode with std::tuple to avoid name confusion

### CODE REVIEW

**Comment by wconstab** [comment]
@desertfire was the tensorpipe submodule change accidental?

**Comment by desertfire** [comment]
> @desertfire was the tensorpipe submodule change accidental?

You are right. Fixed in the new commit.

File: tools/codegen/gen_lazy_tensor.py [hunk 1]
```diff
@@ -200,6 +200,7 @@ def concat_map_codegen(func: Callable[[NativeFunction], Sequence[str]],
                 "torch/csrc/lazy/core/ir.h",
                 "lazy_tensor_core/csrc/ts_backend/ts_node_lowering.h",
                 "torch/csrc/lazy/core/hash.h",
+                "lazy_tensors/str_join.h",
```
Comment (wconstab) [inline tools/codegen/gen_lazy_tensor.py:203]: Is this easy to avoid?  @alanwaketan was replacing all of these with c10::Str(a, b) i think.

**Review by wconstab — APPROVED** [review]
LGTM!

File: tools/codegen/gen_lazy_tensor.py (hunk 1 above)
Comment (desertfire) [reply tools/codegen/gen_lazy_tensor.py:203]: StrJoin was already gone. What's left in str_join.h is a bunch of ToString methods, which are mostly only used in the generated LazyLazyIr.h.

I will do the clean up in another PR. Maybe we could just codegen those methods in LazyLazyIr.h

**Pull request merged**